### ISSUE TITLE
lxc-attach doesn't allow to set uid / gid

### ISSUE THREAD

**Issue by Disassembler0** [lxc/lxc#2591]
# Required information

 * Distribution: Alpine
 * Distribution version: 3.8
 * `lxc-start --version`: 3.0.0+, compiled from git master, commit a21ed5555dfc9bbe8bf1fe3004b3878b6e668f94
* `lxc-checkconfig`: N/A
* `uname -a`: Linux vm 4.14.52-0-virt #1-Alpine SMP Tue Jun 26 07:21:56 UTC 2018 x86_64 Linux
* `cat /proc/self/cgroup`:
```
13:pids:/
12:hugetlb:/
11:net_prio:/
10:perf_event:/
9:net_cls:/
8:freezer:/
7:devices:/
6:memory:/
5:blkio:/
4:cpuacct:/
3:cpu:/
2:cpuset:/
1:name=openrc:/
0::/
```

# Issue description
This is probably more a feature request since the functionality was never there, but here goes.

I'd like lxc-attach to be able to switch UID / GID in the entered namespace. Basically a rough equivalent of `docker exec`, which sets UID to the one given in Dockerfile (unless overridden by `-u`). I assume that `lxc-attach` does pretty much the same as `nsenter`, which allows for options `-S` to set UID and `-G` to set GID for the program subsequently executed in the entered namespace.

My use case is that I have bunch of application containers. One container is running something what needs to be accessible for the others at all times (e.g. postgres). When I'm setting up another container with an application which is supposed to use that postgres, I'm initializing a new database for it from the LXC host. However my postgres setup requires that all actions are done by postgres user, so when I `lxc-attach`, I keep being root and I either need to switch inside the container (via `su`, `sudo`, `runas`, `setuidgid`, `s6-setuidgid` etc.) or I simply can't use `lxc-attach` and have to use `nsenter` instead which allows me to set UID. Also `lxc-execute`, which would normally set UID given by `lxc.init.uid` config value is out of the question, because the container is already running and needs to keep running.

**tl;dr:** Please add set UID / GID switches to `lxc-attach` in similar fashion like `nsenter` or `lxc-execute` have.  
(additionally, `lxc-attach` could honor `lxc.init.uid`, but I'd be happy just with the switches.)

**Comment by brauner** [comment]
Yeah, that should not be difficult to add. Just need to see what flags to use.

**Comment by Disassembler0** [comment]
I have made an attempt to implement the functionality, but I'm not a C programmer and some of the internals are also still a mystery for me... so I have pretty much no idea what I'm doing. :) Before I even try to open it as a PR, I'd like to have a small code review.

The commit is https://github.com/Disassembler0/lxc/commit/b0b1448dc7dcc782e1984358fd2d7ef34ae46a9e

I'm maily concerned about the legality of the rearrangement I've made in *attach.c* where I have moved the `lxc_switch_uid_gid()` call below setgroups, LSM labelling and seccom profile load. If I didn't do that, I'd get *Operation not permitted* on setgroups and *Permission denied* on seccomp policy load. The rest of the code is borrowed from *lxc_execute.c*.

**Comment by brauner** [comment]
What you did looks fine to me. Send it as a signed-off pr. Otherwise I can't give it a proper review. If all is fine I'll merge today or tomorrow.
